src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/CheckAndPublish.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.WebV2.Security;
+using FormFlow;
+using MediatR;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.WebV2.Features.Venues.AddVenue.CheckAndPublish
+{
+    public class Query : IRequest<ViewModel>
+    {
+    }
+
+    public class ViewModel
+    {
+        public string Name { get; set; }
+        public IReadOnlyCollection<string> AddressParts { get; set; }
+        public string Email { get; set; }
+        public string PhoneNumber { get; set; }
+        public string Website { get; set; }
+        public bool NewAddressIsOutsideOfEngland { get; set; }
+    }
+
+    public class Command : IRequest<Success>
+    {
+    }
+
+    public class Handler :
+        IRequestHandler<Query, ViewModel>,
+        IRequestHandler<Command, Success>
+    {
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ICurrentUserProvider _currentUserProvider;
+        private readonly IClock _clock;
+        private readonly IProviderContextProvider _providerContextProvider;
+        private readonly JourneyInstance<AddVenueJourneyModel> _journeyInstance;
+
+        public Handler(
+            ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
+            JourneyInstanceProvider journeyInstanceProvider,
+            ICurrentUserProvider currentUserProvider,
+            IClock clock,
+            IProviderContextProvider providerContextProvider)
+        {
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _currentUserProvider = currentUserProvider;
+            _clock = clock;
+            _providerContextProvider = providerContextProvider;
+            _journeyInstance = journeyInstanceProvider.GetInstance<AddVenueJourneyModel>();
+        }
+
+        public Task<ViewModel> Handle(Query request, CancellationToken cancellationToken)
+        {
+            ThrowIfFlowStateNotValid();
+
+            var state = _journeyInstance.State;
+
+            var addressParts = new[]
```

Comment:
Any particular reason for this variable?

Reply:
I find it easier to parse being separate from the `ViewModel` initialization.